units/provisioning_agent_deploy.go

```diff
@@ -150,6 +150,15 @@ func (j *agentDeployJob) Run(ctx context.Context) {
 	}
 
 	if stat.LastAttemptFailed() && stat.AllAttemptsFailed() && stat.Count >= agentPutRetries {
+		if j.host.Provider != evergreen.ProviderNameStatic {
```

Comment:
Can this check happen in `CheckExternallyTerminatedHost`?

Reply:
done.